### ISSUE TITLE
How to pass multiple images as input？

### ISSUE THREAD

**Issue by brxhjh** [QwenLM/Qwen2.5-VL#1179]
I want to input two images but I don't know how to achieve it.
I have write a coding as follows,but it will go wrong that pyarrow.lib.ArrowInvalid: cannot mix list and non-list, non-null values
So I think my images may input wrong?
thanks for help!!

coding:
def process_func(example):

    MAX_LENGTH = 8192   
    conversation = example["conversations"]
    input_content = conversation[0]["value"]
    output_content = conversation[1]["value"]

    image_paths = example["image"]  # 这应该是一个列表，可能包含1个或2个图像路径#image_paths is a list which contains one or two images
    
    messages = [
        {
            "role": "user",
            "content": [],
        }
    ]
    
    # 添加所有图像到content
    for img_path in image_paths:
        messages[0]["content"].append(
            {"type": "image", "image": f"{img_path}", "resized_height": 224, "resized_width": 224}
        )
    
    # 添加文本内容
    messages[0]["content"].append(
        {"type": "text", "text": f"{input_content}"}
    )
    
    text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
    image_inputs, video_inputs = process_vision_info(messages)
    inputs = processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=True,
        return_tensors="pt",
    )

    inputs = {key: value.tolist() for key, value in inputs.items()}
    
    # 构造目标输出
    response = tokenizer(f"{output_content}", add_special_tokens=False)
    input_ids = inputs["input_ids"][0] + response["input_ids"] + [tokenizer.pad_token_id]
    attention_mask = inputs["attention_mask"][0] + response["attention_mask"] + [1]
    labels = [-100] * len(inputs["input_ids"][0]) + response["input_ids"] + [tokenizer.pad_token_id]

    # 截断
    if len(input_ids) > MAX_LENGTH:
        input_ids = input_ids[:MAX_LENGTH]
        attention_mask = attention_mask[:MAX_LENGTH]
        labels = labels[:MAX_LENGTH]

    return {
        "input_ids": torch.tensor(input_ids),
        "attention_mask": torch.tensor(attention_mask),
        "labels": torch.tensor(labels),
        "pixel_values": torch.tensor(inputs["pixel_values"]),
        "image_grid_thw": torch.tensor(inputs["image_grid_thw"]).squeeze(0)
    }

**Comment by aaaalonga** [comment]
Same question，did you solve it?
